addons/l10n_ar_withholding/models/account_payment.py

```diff
@@ -0,0 +1,23 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models
+
+
+class AccountPayment(models.Model):
+
+    _inherit = 'account.payment'
+
+    def _synchronize_to_moves(self, changed_fields):
+        ''' If we change a payment with withholdings, delete all withholding lines as the synchronization mechanism is not
+        implemented yet
+        '''
+        if self._context.get('skip_account_move_synchronization'):
+            return
+
+        if not any(field_name in changed_fields for field_name in self._get_trigger_fields_to_synchronize()):
+            return
+
+        for pay in self.with_context(
+                skip_account_move_synchronization=True, check_move_validity=False, skip_invoice_sync=True, dynamic_unlink=True):
```

Comment:
Check `check_move_validity` if needed to be here with False
@jjscarafia @maq-adhoc